lib/process/watch.go

```diff
@@ -35,6 +41,100 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+// runCertExpirationWatch checks if the default self signed cluster cert is about to expire soon and updates it
+func (p *Process) runCertExpirationWatch(client *kubernetes.Clientset) clusterService {
+	return func(ctx context.Context) {
+		ticker := time.NewTicker(time.Hour * 24)
+		defer ticker.Stop()
+		for {
+			err := p.replaceCertIfAboutToExpire(ctx, client)
+			if err != nil {
+				p.WithError(err).Error("Failed to check for certificate expiration or replace it.")
+			}
+
+			select {
+			case <-ticker.C:
+			case <-ctx.Done():
+				p.Debug("Certificate expiration watcher stopped.")
+				return
+			}
+		}
+	}
+}
+
+func (p *Process) replaceCertIfAboutToExpire(ctx context.Context, client *kubernetes.Clientset) error {
+	p.Info("Running self signed certificate expiration check/rotation...")
+
+	ticker := backoff.NewTicker(&backoff.ExponentialBackOff{
+		InitialInterval: time.Second * 3,
+		Multiplier:      1.5,
+		MaxInterval:     10 * time.Second,
+		MaxElapsedTime:  1 * time.Minute,
+		Clock:           backoff.SystemClock,
+	})
+	defer ticker.Stop()
+
+	for {
+		select {
+		case tm := <-ticker.C:
+			if tm.IsZero() {
+				return trace.ConnectionProblem(nil, "timed out waiting while checking for certificate expiration")
+			}
+			clusterCert, _, err := opsservice.GetClusterCertificate(client)
+			if err != nil {
+				p.WithError(err).Error("Failed to retrieve the certificate from k8s.")
```

Comment:
In the log messages here and below, I would specify what certificate is being referred to exactly, otherwise it may not be clear from the logs. I.e. "cluster web UI certificate" or something like this.